azurerm/resource_arm_app_service.go

```diff
@@ -709,12 +755,33 @@ func flattenAppServiceSiteConfig(input *web.SiteConfig) []interface{} {
 	return append(results, result)
 }
 
-func flattenAppServiceSourceControl(input *web.SiteSourceControlProperties) []interface{} {
+func expandAppServiceSourceControl(d *schema.ResourceData) web.SiteSourceControl {
+	scmList := d.Get("source_control").([]interface{})
+	scmRet := web.SiteSourceControl{}
+
+	if len(scmList) == 0 {
+		return scmRet
+	}
+
+	scm := scmList[0].(map[string]interface{})
+
+	if v, ok := scm["repo_url"]; ok {
+		scmRet.RepoURL = utils.String(v.(string))
+	}
+
+	if v, ok := scm["branch"]; ok {
+		scmRet.Branch = utils.String(v.(string))
+	}
+
+	return scmRet
+}
+
+func flattenAppServiceSourceControl(input *web.SiteSourceControl) []interface{} {
```

Comment:
we should be pulling from the `SiteSourceControlProperties` object since the non-Properties object isn't actually valid in Azure (e.g. it doesn't actually match the schema and can be nil in older configurations)

Reply:
Internal Go SDK is using SiteSourceControl as returned type and SiteSourceControlProperties is embedded class, which means that you can access all the fields in SiteSourceControl. It's meaningless to do any kind of de-referencing before flattening.

I don't understand what's your concern here, maybe a more concrete example ?

Reply:
at this point in time, it's changed in the past - that's not how it's represented in the JSON Response so we should match that where possible (which we do throughout the provider)

Reply:
There's no any API return SiteSourceControlProperties, instead, it's SiteSourceControl.
If SiteSourceControl is nil, then you will not be able to get SiteSourceControlProperties, since they're the same object but different layer.